### ISSUE TITLE
Playback dynamic seemingly randomly changes without a change in dynamic mark.

### ISSUE THREAD

**Issue by matthewreadbass** [musescore/MuseScore#19409]
### Issue type

General playback bug

### Bug description

I found that the dynamic of passages occasionally seems to reset to MF without a change in dynamic marking.

Weirdly, this isn't a consistent change for me and I have only noticed it happening today despite having been using the attached testing template for ~1 week.

This issue seemingly only happens at the end of a phrase and appears to be triggered by either a rest or a double bar line (although removing the double bar line didn't fix the issue - I mention it as it is one of only two new elements when the reset happens).

### Steps to reproduce

Hopefully this will work for you, but as it is inconsistent for me I don't know if it will...

1. Unzip and open attached file
2. play back from 4-8 bars before a double bar line/rest
3. The dynamic might reset from FFF to MF...

[Trombone Rhythm test.mscz.zip](https://github.com/musescore/MuseScore/files/12619209/Trombone.Rhythm.test.mscz.zip)

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/80341862/2720ace0-e8cd-4ced-a3f1-20614e5eb8cc

### MuseScore Version

4.1.1

### Regression

I don't know

### Operating system

mac OS 13.5

**Comment by bkunda** [comment]
Does this happen with soundfonts and VSTs, or only Muse Sounds?

**Comment by matthewreadbass** [comment]
I only got this to happen a handful of times and it doesn't seem reliably re-creatable. As such, I am yet to confirm whether this happens with soundfonts/VSTs

**Comment by zacjansheski** [comment]
I haven't been able to reproduce this yet.  The attached score does get much softer on the last line of the first page for me.

Just to confirm: this was made from scratch in MS4?  It wasn't imported?

**Comment by matthewreadbass** [comment]
It was indeed made from scrstch

**Comment by zacjansheski** [comment]
Perhaps this is a Muse Sounds issue, changing to MS Basic seems to get rid of the dynamic change

https://github.com/musescore/MuseScore/assets/69917893/146b0295-6181-40c3-a075-400af279229b

**Comment by zacjansheski** [comment]
The ppp later on in the score seems to be the focal point of the issue.  Deleting it removes the change in dynamic

https://github.com/musescore/MuseScore/assets/69917893/dd68e832-389f-4ce2-904f-4547b88f751d

**Comment by msviolafangirl** [comment]
I agree that this seems to be a Muse Sounds issue. I've noticed it happening with Muse Sounds for seemingly no reason and without any real consistency, so it is harder to prove. MS basic does not have this issue.

**Comment by Valdessir** [comment]
Likely related to #21522
